neuralop/models/base_model.py

```diff
@@ -76,7 +76,33 @@ def __new__(cls, *args, **kwargs):
         instance._init_kwargs = kwargs
 
         return instance
-    
+
+    def state_dict(self, **kwargs):
+        """
+        state_dict subclasses nn.Module.state_dict() and adds a metadata field
+        to track the model version and ensure only compatible saves are loaded.
+        """
+        print("In state dict")
+        state_dict = super().state_dict(**kwargs)
+        state_dict.update({'neuralop_metadata': self._init_kwargs})
+        return state_dict
+
+    def load_state_dict(self, state_dict, **kwargs):
+        """
+        state_dict subclasses nn.Module.load_state_dict() and adds a metadata field
+        to track the model version and ensure only compatible saves are loaded.
+        """
+        metadata = state_dict.get('neuralop_metadata')
```

Comment:
Is this tested? I believe get returns None by default so the `if metadata` below would fail.
Let's make the default `None` explicit and test `if metadata is not None` below to remove the ambiguity.

Let's also use pop rather than get so we don't need to remove the metadata before passing to super().

Reply:
Sounds good.

Reply:
Note that `pop` will throw a keyerror though.